fixtures/coverall/tests/bindings/test_coverall.py

```diff
@@ -75,7 +75,7 @@ def test_constructors(self):
         with self.assertRaises(CoverallError.TooManyHoles):
             Coveralls.fallible_new("", True)
 
-        with self.assertRaisesRegex(InternalError, "expected panic: woe is me"):
+        with self.assertRaisesRegex(InternalError, "Rust panic"):
```

Comment:
Oops, meant to flag this but I forgot.  I was worried about what happens if the code to extract the string itself panics.  For example, it seems like the current code could panic: https://github.com/mozilla/ffi-support/blob/31ffbbd6eec90038d825dab145c5382ca248b2f4/src/error.rs#L257.  I think calling `to_string()` on a `str` shouldn't panic.  But couldn't calling `clone()` on a `String` panic when trying to allocate the new string?  There's also the issue of converting the `String` to a `RustBuffer`, which I think should be panic-free, but I'm not sure about that.

I was hoping that the panic hook could log enough info to debug things, maybe as a Sentry breadcrumb.  I'm not sure if this will work though.  (BTW, what happens if the panic hook panics?  I couldn't find any documentation on this).

If that's not possible, then I think we should put the code to extract the panic string in a second `catch_unwind` block.

Reply:
Yes, you're right, I believe at least the allocation underlying `String::clone` could fail in this case and trigger a panic.

> I was hoping that the panic hook could log enough info to debug things, maybe as a Sentry breadcrumb.

We haven't had good luck getting rust-level debugging output into Sentry, although it's probably time we tried again. 

> I think we should put the code to extract the panic string in a second catch_unwind block.

Making a best-effort attempt to get the string out along these lines seems valuable to me 👍🏻